### PR TITLE
benchmark: add pfm option

### PR DESCRIPTION
**Pull request by andioz**
Specify library name and version:  **benchmark/1.8.0**

Benchmark has a build option for enabling performance counters: https://github.com/google/benchmark/blob/v1.5.4/docs/perf_counters.md

We want to use this feature, therefore I try to add an option which is disabled by default. As far as I have seen this is currently only for Linux supported, and requires the "libpfm4" library, which is available via Conan. This feature was added with version 1.5.4, therefore I include a version check too (even if our supported versions starts with a higher version number). 

Some time past by since I made my last contribution, and this is my very first in the time of Conan 2.x - I hope I did everything in the right way, please correct me if not.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by Hopobcn — APPROVED** [review]

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Requesting changes until the linked PR is merged so this gets the latest revision :)

**Comment by andioz** [comment]
> Requesting changes until the linked PR is merged so this gets the latest revision :)

Do you plan to update the libpfm version too? (4.13.0 is available)

**Comment by RubenRBS** [comment]
Ah I missed that, yes! Will do

File: recipes/benchmark/all/conanfile.py [hunk 1]
```diff
@@ -49,6 +53,10 @@ def validate(self):
         if Version(self.version) < "1.7.0" and is_msvc(self) and self.options.shared:
             raise ConanInvalidConfiguration(f"{self.ref} doesn't support msvc shared builds")
 
+    def requirements(self):
+        if self.options.get_safe("enable_libpfm"):
+            self.requires("libpfm4/4.12.0")
```
Comment (RubenRBS) [inline recipes/benchmark/all/conanfile.py:58]:
```suggestion
            self.requires("libpfm4/4.13.0")
```

**Review by RubenRBS — CHANGES_REQUESTED** [review]

**Review by RubenRBS — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**